python/samples/demos/guided_conversations/guided_conversation/utils/plugin_helpers.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from dataclasses import dataclass
+
+from pydantic import ValidationError
+from semantic_kernel import Kernel
+from semantic_kernel.connectors.ai.open_ai import AzureChatCompletion
+from semantic_kernel.connectors.ai.open_ai.services.utils import kernel_function_metadata_to_openai_tool_format
+from semantic_kernel.functions import KernelArguments
+
+from guided_conversation.utils.conversation_helpers import ConversationMessage
+from guided_conversation.utils.openai_tool_calling import parse_function_result, validate_tool_calling
+
+
+@dataclass
+class PluginOutput:
+    """A wrapper for all Guided Conversation Plugins. This class is used to return the output of a generic plugin.
+
+    Args:
+        update_successful (bool): Whether the update was successful.
+        messages (list[ConversationMessage]): A list of messages to be used at the user's digression, it
+        contains information about the process of calling the plugin.
+    """
+
+    update_successful: bool
+    messages: list[ConversationMessage]
+
+
+def format_kernel_functions_as_tools(kernel: Kernel, functions: list[str]):
+    """Format kernel functions as JSON schemas for custom validation."""
+    formatted_tools = []
+    for _, kernel_plugin_def in kernel.plugins.items():
+        for function_name, function_def in kernel_plugin_def.functions.items():
+            if function_name in functions:
+                func_metadata = function_def.metadata
+                formatted_tools.append(kernel_function_metadata_to_openai_tool_format(func_metadata))
```

Comment:
This should now be `kernel_function_metadata_to_function_call_format()`: https://github.com/microsoft/semantic-kernel/blob/c15e253097c6805752cdc62ade4e05ccce290e3a/python/semantic_kernel/connectors/ai/function_calling_utils.py#L28

Reply:
Done!